### DISCUSSION TITLE
Cannot find module '@coder/logger'

### DISCUSSION THREAD

**Discussion by WeAreAllLonlyGhost** [coder/code-server#4878] [Q&A] [answered]
I`m trying to migrate code-server from one docker image to annother through mounting.But when i run code-server i get this error.
![image](https://user-images.githubusercontent.com/59404750/154454872-06fc17c9-c3cc-47f8-af0b-c42600f43f6d.png)

i mounted such files to the new image
![image](https://user-images.githubusercontent.com/59404750/154460585-71010d8c-e39e-44f8-a55a-7276fbb020b9.png)

**Comment by lg-nineoneninethreenine** [comment]
Hmm, are you using GCP? Have you tried making a Git repo to transfer the files? That might work...

**Reply by jsjoeio** [reply]
Hmm...I'm not sure how that should work (migrating to another docker image). Ideally you shouldn't have to mount these directories

**Reply by WeAreAllLonlyGhost** [reply]
What i'm trying to do is mounting code-server to image so that i can use the same image through different porject,and the base image won't be too big.Don't know if it's gonna work....

**Reply by lg-nineoneninethreenine** [reply]
It could be the wrong directory of the code-server modules. I'll try unzipping a release tarball to check.

**Comment by lg-nineoneninethreenine** [comment]
Ladies and gentlemen, I guess we have an answer. This:
![image](https://user-images.githubusercontent.com/76845820/154691019-800291cb-b9c9-4fd2-928d-e02666504b80.png)
It _has_ to be the `$PATH` of the node modules folder. @WeAreAllLonlyGhost try changing the DIR name in your Docker config.

**Comment by lg-nineoneninethreenine** [comment]
Wait, did you use `npm` to install code-server? Firstly, what _are_ you trying to move? `code-server`'s configurations(~/.config/code-server/config.yaml`) or the extensions, settings, user data, etc? I think for extensions, you make a folder where the VSIX files reside, so it's easy for switching them over. You can copy the config from one place to another(don't reveal the password unless it's a localhost connection) as well as access the JSON version of the settings and copy it over to the other side.

**Reply by WeAreAllLonlyGhost** [reply]
That's the thing.I don't want to use npm install,just want to cpoy those related files and start code-server.For example ,i install code-ser on my host machine(~/.local/*),then i just copoy those files into container and start it.(don't konw if it's possible)

**Reply by lg-nineoneninethreenine** [reply]
Instead, try using code-server tarballs. They have binaries for all Linux distros hence I recommend that you use it. Remember, if you use ARM64, choose the aarch64 or arm64 version of the tarball.

**Comment by WeAreAllLonlyGhost** [comment, marked as answer]
Solved!Beacause i use .zip to move those files from image,some link files were zipped to actually entity files!

**Reply by lg-nineoneninethreenine** [reply]
Well, be careful from now!

**Reply by jsjoeio** [reply]
Phew! Glad it was an easy fix!
